course_contents/2_intro_to_python/notes/04. Destructuring.md

```diff
@@ -0,0 +1,49 @@
+### Unpacking
+- unpacking is used to assign values from a collection to variables in a single statement
+- syntax for unpacking:
+
+```py
+x, y, z = collection
+```
+
+- the collection can be a list, tuple, string, etc.
+
+An example of unpacking a list:
+
+```py
+example_list = [1,2,3]
+a, b, c = example_list
+print(a, b, c)
+```
+
+### The `enumerate()` method
+- the `enumerate()` method is used to retrieve an object containing both element and its index inside the collection that is being iterated over
+- syntax for the `enumerate()` method with unpacking:
+
+```py
+for index, element in enumerate(collecion):
+    block of code
+```
+
+An example of the `enumerate()` method:
+
+```py
+collection = ['a', 'b', 'c']
+for index, element in enumerate(collection):
+    print(f"Index:{index}; Element:{element}")
+```
+
+### The `zip()` function
+- the `zip()` function is used to combine two iterables into a single one
+- new iterable is created by joining i-th element of each iterable until the end of the shorter one is reached
+- the `zip()` function returns a `zip` object
+- syntax for the `zip()` function:
+
+An example of the `zip()` function used on two lists:
+
+```py
+numbers = [1, 2, 3, 4, 5]
+letters = ['a', 'b', 'c']
+combined = list(zip(numbers, letters))
+print(combined)
```

Comment:
Could you come up with something better than `numbers` and `letters`?

Reply:
I think there is no need to, as I mentioned in the above comments since the sole purpose of these notes is for students to be able to quickly refresh their memory of terms that they've heard about in the course. Again, if Jose wants, I can create more complex examples that would look more like real problems